### PR TITLE
Buffs Lightning Bolt

### PR DESCRIPTION
**Pull request by AnturK**
Lightning Bolt now : 
- Always bounces 5 times
- Does from 15 to 50 damage on each target depending on the chargeup
- Has the charge-bar
- Affects people with insulated gloves (that was unintended)

### CODE REVIEW

**Comment by Xhuis** [comment]
This has been needed for so long :+1:

**Comment by Durkel** [comment]
Good change.

File: code/datums/spells/lightning.dm [hunk 1]
```diff
@@ -72,26 +70,18 @@
...
-			target.electrocute_act(10,"Lightning Bolt")
-			playsound(get_turf(target), 'sound/magic/LightningShock.ogg', 50, 1, -1)
-		if(25 to 75)
-			target.electrocute_act(25,"Lightning Bolt")
-			playsound(get_turf(target), 'sound/magic/LightningShock.ogg', 50, 1, -1)
-		if(75 to 100)
-			//CHAIN LIGHTNING
-			Bolt(user,target,energy,user)
+	var/energy = min(world.time - start_time,100)
+	Bolt(user,target,max(15,energy/2),5,user) //5 bounces for energy/2 burn
```
Comment (Dorsisdwarf) [inline code/datums/spells/lightning.dm]: Why do you need to input user as a parameter twice?

Comment (AnturK) [inline code/datums/spells/lightning.dm]: First one is source of bolt, second is spellcaster.

Comment (Anonus) [inline code/datums/spells/lightning.dm]: Wouldn't those be the same guy like 99% of the time, or am I missing something?

Comment (AnturK) [inline code/datums/spells/lightning.dm]: Wiz :zap: Victim A :zap: Victim B :zap: Victim C
Bolt(Wiz,A,Wiz)
Bolt(A,B,Wiz)
Bolt(B,C,Wiz)

Comment (Dorsisdwarf) [inline code/datums/spells/lightning.dm]: But both of them are user and called at the same time, presumably therefore inputting the same value unless the variable is set to both at the same time?

Comment (AnturK) [inline code/datums/spells/lightning.dm]: I'm not sure what you mean here. Bolt is recursive. This is just to keep track of caster and don't zap him on rebound, i probably could get away with using usr, but i prefer stuff like this to be explicit.

**Comment by WJohn** [comment]
Definitely needs changelog.

Good change :+1:

**Comment by AnturK** [comment]
Oh right. Will add changelog tommorow.

File: code/datums/spells/lightning.dm (hunk 1 above)
Comment (Dorsisdwarf) [inline code/datums/spells/lightning.dm]: This isn't saying "amg this is awful remuv", this is me wondering how an interesting bit of code works.

**Comment by Steelpoint** [comment]
I had a chance to play with lightning bolt on the servers and I do think this change is needed.

**Pull request merged**